mlrun/projects/project.py

```diff
@@ -1647,7 +1648,19 @@ def push(self, branch, message=None, update=True, remote=None, add: list = None)
         if repo.is_dirty():
             if not message:
                 raise ValueError("please specify the commit message")
-            repo.git.commit(m=message)
+            try:
+                repo.git.commit(m=message)
+            except git.exc.GitCommandError as exc:
+                if "Please tell me who you are" in str(exc):
+                    warning_message = (
+                        "Git is not configured, please run the following commands and try again:\n"
+                        '\tgit config --global user.email "<my@email.com>"\n'
+                        '\tgit config --global user.name "<name>"\n'
+                        "\tgit config --global credential.helper store\n"
+                    )
+                    logger.error(warning_message)
+                    return
```

Comment:
when `Please tell me who you` in err it seems like no error will be raised to the user but rather only a log message, I think we would like to raise as before